File: Source/ExpressionValidator.cpp
#include "ExpressionValidator.h"

ExpressionValidator::ExpressionValidator() : m_regex("^(\\(?\\d+\\s?[\\+\\-\\*\\/]\\s?)+\\d+\\)?$")
{
}

bool ExpressionValidator::Validate(const std::string& expression)
{
	return std::regex_match(expression, m_regex);
}

File: src/main/java/vsb/gai0010/ExpressionValidator.java
package vsb.gai0010;

import java.util.regex.Pattern;

public class ExpressionValidator {
    final private Pattern pattern;

    public ExpressionValidator() {
        pattern = Pattern.compile("^(\\(?\\d+\\s?[+\\-*/]\\s?)+\\d+\\)?$");
    }

    public boolean validate(String expression) {
        return this.pattern.matcher(expression).find();
    }
}
